instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/src/main/java/io/opentelemetry/instrumentation/auto/jaxrs/v2_0/JaxRsAnnotationsInstrumentation.java

```diff
@@ -155,15 +157,23 @@ public static void stopSpan(
         return;
       }
 
+      CompletionStage<?> asyncReturnValue =
+          returnValue instanceof CompletionStage ? (CompletionStage<?>) returnValue : null;
+
       if (asyncResponse != null && !asyncResponse.isSuspended()) {
         // Clear span from the asyncResponse. Logically this should never happen. Added to be safe.
         InstrumentationContext.get(AsyncResponse.class, Span.class).put(asyncResponse, null);
       }
-      if (asyncResponse == null || !asyncResponse.isSuspended()) {
+      if (asyncReturnValue != null) {
+        // span finished by CompletionStageFinishCallback
+        asyncReturnValue = asyncReturnValue.whenComplete(new CompletionStageFinishCallback<>(span));
```

Comment:
I don't think we consistently do so in this codebase, but FWIU `handle` has much better performance than `whenComplete` (due to a JDK bug?)

https://github.com/line/armeria/issues/1426

Can we use handle here?

Reply:
> Can we use handle here?

Of course -- changed. Thanks!